src/sortStudents.ts

```diff
@@ -1,16 +1,94 @@
+/* eslint-disable no-debugger */
+/* eslint-disable no-console */
+/* eslint-disable max-len */
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  let result: Student[] = [];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function CompareFn(a: Student, b: Student): number {
+    let retValue: number = 0;
+
+    switch (sortBy) {
+      case SortType.Name:
```

Comment:
A lot of things seem to be repeating. You can improve the code by combining cases with similar type 
```suggestion
      case SortType.Name:
      case SortType.Surname:
```